private/protocol/rest/build.go

```diff
@@ -186,7 +186,12 @@ func buildHeaderMap(header *http.Header, v reflect.Value, tag reflect.StructTag)
 		keyStr := strings.TrimSpace(key.String())
 		str = strings.TrimSpace(str)
 
-		header.Add(prefix+keyStr, str)
+		if normalize {
+			lk := strings.ToLower(prefix + keyStr)
+			(*header)[lk] = append((*header)[lk], str)
```

Comment:
Nit: This seems complicated to understand right away, why not use `Get, Set` methods of header?